### PR TITLE
[traffic-gen-in-vm] Checkup: remove traffic-gen pod

### PR DESCRIPTION
**Pull request by orelmisan**
In preparation for the conversion of the traffic generator pod to a VMI, remove all traffic-gen code from the checkup:
1. The logic to set up and teardown the pod.
2. The client code responsible for Pod and NetworkAttachmentDefinition operations.
3. RBAC rules for Pod and NAD operations.
4. Traffic-gen's SA and SCC.

Also update the README file.

Follow-up PRs will remove:
1. Unused configuration fields.
2. Traffic-gen pod build system.

~~Depends on #118.~~

Tested against an OpenShift Virtualization 4.14 cluster.

### CODE REVIEW

File: pkg/internal/checkup/executor/executor.go [hunk 1]
```diff
@@ -41,307 +35,26 @@ type vmiSerialConsoleClient interface {
...
-		vmiEastNICPCIAddress:             config.VMIEastNICPCIAddress,
-		vmiEastEthPeerMACAddress:         cfg.TrafficGeneratorEastMacAddress.String(),
-		vmiWestNICPCIAddress:             config.VMIWestNICPCIAddress,
-		vmiWestEthPeerMACAddress:         cfg.TrafficGeneratorWestMacAddress.String(),
-		testDuration:                     cfg.TestDuration,
-		verbosePrintsEnabled:             cfg.Verbose,
-		trafficGeneratorPacketsPerSecond: cfg.TrafficGeneratorPacketsPerSecond,
+		client:      client,
+		namespace:   namespace,
+		vmiUsername: config.VMIUsername,
```
Comment (RamLavi) [inline pkg/internal/checkup/executor/executor.go:49]: why do you keep LoginToCentOS? We will need to revise that as well in future commits.

Comment (orelmisan) [reply pkg/internal/checkup/executor/executor.go:49]: It is the minimal usage of the serial console, I don't think it needs to be removed (but will probably be changed).
I don't see how we can not login to the guests.

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/82db4db07349ec44fc89016ff6706b57d89d59be..9ba9d5ff74f1d9f2657964e60a33a55c78e225a7): Rebase

File: tests/checkup_test.go [hunk 2]
```diff
@@ -105,8 +105,6 @@ var _ = Describe("Execute the checkup Job", func() {
 		Expect(configMap.Data).NotTo(BeNil())
 		Expect(configMap.Data["status.succeeded"]).To(Equal("true"), fmt.Sprintf("should succeed %+v", prettifyData(configMap.Data)))
 		Expect(configMap.Data["status.failureReason"]).To(BeEmpty(), fmt.Sprintf("should be empty %+v", prettifyData(configMap.Data)))
-		Expect(configMap.Data["status.result.trafficGeneratorNode"]).ToNot(BeEmpty(),
```
Comment (RamLavi) [inline tests/checkup_test.go:108]: why is this removed?

Comment (orelmisan) [reply tests/checkup_test.go:108]: Because there is no traffic generator pod, so it is not scheduled to any node.

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/9ba9d5ff74f1d9f2657964e60a33a55c78e225a7..6aeb8c2c2d3b9cd4a4de8cfd803e4f91e62d5cc5): Rebase

**Comment by orelmisan** [comment]
E2E test passed after the rebase.

**Comment by RamLavi** [comment]
> * The checkup logic from executor.

Can you update the PR desc to remove this point?

File: pkg/internal/checkup/checkup.go [hunk 3]
```diff
@@ -156,18 +156,10 @@ func (c *Checkup) Teardown(ctx context.Context) error {
 		return fmt.Errorf("%s: %w", errPrefix, err)
 	}
 
-	if err := c.deletePod(ctx); err != nil {
```
Comment (RamLavi) [inline pkg/internal/checkup/checkup.go:159]: what about the actual implementation? Don't you want to remove it as well?

File: pkg/internal/checkup/checkup.go [hunk 4]
```diff
@@ -259,134 +233,6 @@ func (c *Checkup) cleanupVMI() {
...
-	if err := wait.PollImmediateUntilWithContext(ctx, interval, conditionFn); err != nil {
-		return fmt.Errorf("failed to wait for Pod '%s' to be in Running Phase: %v", podFullName, err)
-	}
-
-	log.Printf("Pod %s is Running", podFullName)
-	c.trafficGeneratorPod = updatedPod
-	return nil
-}
-
-func (c *Checkup) deletePod(ctx context.Context) error {
```
Comment (RamLavi) [inline pkg/internal/checkup/checkup.go:348]: shouldn't this be deleted in the former commit?

File: pkg/internal/checkup/checkup.go [hunk 5]
```diff
@@ -259,134 +233,6 @@ func (c *Checkup) cleanupVMI() {
...
-	log.Printf("Trying to delete traffic generator Pod: %q", vmiFullName)
-	if err := c.client.DeletePod(ctx, c.trafficGeneratorPod.Namespace, c.trafficGeneratorPod.Name); err != nil {
-		log.Printf("Failed to delete traffic generator Pod: %q", vmiFullName)
-		return err
-	}
-
-	return nil
-}
-
-func (c *Checkup) waitForPodDeletion(ctx context.Context) error {
```
Comment (RamLavi) [inline pkg/internal/checkup/checkup.go:364]: ditto

**Comment by orelmisan** [comment]
> > * The checkup logic from executor.
> 
> Can you update the PR desc to remove this point?

Done.

**Review by RamLavi — CHANGES_REQUESTED** [review]
Hey, added some comment. 
I also think we need to remove [this](https://github.com/kiagnose/kubevirt-dpdk-checkup/blob/6aeb8c2c2d3b9cd4a4de8cfd803e4f91e62d5cc5/README.md?plain=1#L7) line from the README

File: pkg/internal/checkup/checkup.go (hunk 4 above)
Comment (orelmisan) [reply pkg/internal/checkup/checkup.go:348]: No, because in case there is a setup failure - there is also a cleanup of the pod that uses this method.

File: pkg/internal/checkup/checkup.go (hunk 5 above)
Comment (orelmisan) [reply pkg/internal/checkup/checkup.go:364]: Same answer.

**Comment by orelmisan** [comment]
> Hey, added some comment. I also think we need to remove [this](https://github.com/kiagnose/kubevirt-dpdk-checkup/blob/6aeb8c2c2d3b9cd4a4de8cfd803e4f91e62d5cc5/README.md?plain=1#L7) line from the README

Good catch.
Done.

File: pkg/internal/checkup/checkup.go (hunk 3 above)
Comment (orelmisan) [reply pkg/internal/checkup/checkup.go:159]: It is removed in the second commit.

Comment (RamLavi) [reply pkg/internal/checkup/checkup.go:159]: but it should be removed in the first one, no?

**Review by RamLavi — APPROVED** [review]
/lgtm
/approve
